### PR TITLE
basic part support

### LINKED ISSUE #36: Multiple parts
@jaredjj3 I just pushed a minor commit to support breve and long notes. There are only two tests failing and they are related to multiple parts scored which still not supported.

There are two work arounds for VexFlow bugs documented with 'VexFlow bug:' in the code

### PR DESCRIPTION
**Pull request by rvilarl**
Fixes #36

Only basic support. We should start to divide renderer.ts in different files

### CODE REVIEW

File: src/producer.ts [hunk 1]
```diff
@@ -29,12 +31,36 @@ export class Producer {
         receiver.onMessage(message);
       }
     }
+    while (this.partsIndex[0] < this.parts[0].childs.length) {
+      const messages: EasyScoreMessage[] = [];
+      for (let i = 0; i < this.parts.length; i++) {
+        messages.push({ msgType: 'partStart', id: this.parts[i].id, ndx: i + 1, from: this.parts.length });
```
Comment (jaredjj3) [inline src/producer.ts:37]: nit: It's not an index if you're adding 1 like this. Would you either rename the property as "number" or remove the `+ 1` please?

File: src/producer.ts [hunk 2]
```diff
@@ -29,12 +31,36 @@ export class Producer {
...
 
   private getMessages(node: Node): EasyScoreMessage[] {
     const messages: EasyScoreMessage[] = [];
     const nodeElem = node as Element;
     switch (node.nodeName) {
+      case 'part':
+        {
+          const id = nodeElem.getAttribute('id') ?? 'NN';
+          this.parts.push({ id, childs: node.childNodes });
+          this.partsIndex.push(0);
```
Comment (jaredjj3) [inline src/producer.ts:61]: This doesn't seem to be doing anything. Do we need `this.partsIndex` at all?

File: src/renderer.ts [hunk 3]
```diff
@@ -96,17 +97,69 @@ export class Renderer {
...
+            const notesWidth =
+              staveWidth -
+              staves.get(`${curPart}_1`)!.getNoteStartX() +
+              staves.get(`${curPart}_1`)!.getX() -
+              VF.Stave.defaultPadding;
+            formatter!.format(voicesArr, notesWidth);
+            t.literal(`formatter.format(voicesArr, ${notesWidth});`);
+            t.expression(() => factory.draw());
+            x += staves.get(`${curPart}_1`)!.getWidth();
+            //t.expression(() => voices.clear());
```
Comment (jaredjj3) [inline src/renderer.ts:137]: nit: Remove

File: src/renderer.ts [hunk 4]
```diff
@@ -96,17 +97,69 @@ export class Renderer {
...
+            //t.expression(() => staves.clear());
+            if (x > 1500) {
```
Comment (jaredjj3) [inline src/renderer.ts:139]: It seems like we need to formalize what the context's width should be. Later on, I think we need to let the renderer know what width to use. For now, do you mind lifting all the values like this into named constants so we can see the relationship between all of them more clearly? Feel free to do that in a downstream PR.

File: src/types.ts [hunk 5]
```diff
@@ -4,6 +4,13 @@ export type MeasureStartMessage = {
   staves?: number;
 };
 
+export type PartStartMessage = {
+  msgType: 'partStart';
+  id: string;
+  ndx: number;
+  from: number;
```
Comment (jaredjj3) [inline src/types.ts:11]: nit: Based on its usage, this `from` property references the `ndx`. Would you indicate that in the property name?

File: src/types.ts [hunk 6]
```diff
@@ -16,6 +23,13 @@ export type MeasureEndMessage = {
   msgType: 'measureEnd';
 };
 
+export type PartEndMessage = {
+  msgType: 'partEnd';
+  id: string;
+  ndx: number;
+  from: number;
```
Comment (jaredjj3) [inline src/types.ts:30]: nit: Based on its usage, this `from` property references the `ndx`. Would you indicate that in the property name?

**Review by jaredjj3 — CHANGES_REQUESTED** [review]
IMO, this is a huge improvement. Thanks for adding this.

File: src/types.ts (hunk 5 above)
Comment (rvilarl) [reply src/types.ts:11]: Waht name do you recommend? ndx will be number

Comment (rvilarl) [reply src/types.ts:11]: maxNumber, total, ??

File: src/producer.ts (hunk 2 above)
Comment (rvilarl) [reply src/producer.ts:61]: partsIndex is used above on lines 39 & 43. They are used to keep track on how far we wentwith the messages on each part for each of the measures.

File: src/types.ts (hunk 5 above)
Comment (rvilarl) [reply src/types.ts]: I went for msgIndex and msgCount.

File: src/renderer.ts (hunk 4 above)
Comment (rvilarl) [reply src/renderer.ts:137]: At want stage I will add support for print messages. MusicXML dictates the width of the page.

File: src/producer.ts (hunk 2 above)
Comment (jaredjj3) [reply src/producer.ts:61]: SG

File: src/renderer.ts (hunk 4 above)
Comment (jaredjj3) [reply src/renderer.ts:137]: SG

**Review by jaredjj3 — CHANGES_REQUESTED** [review]
LGTM, thanks for adding this.

**Review by jaredjj3 — APPROVED** [review]

**Pull request merged**